subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -106,28 +106,31 @@ static void isoal_sink_deallocate(isoal_sink_handle_t hdl)
 /**
  * @brief Create a new sink
  *
- * @param hdl[out]        Handle to new sink
- * @param cis[in]         Poiner to iso stream
- * @param sdu_alloc[in]   Callback of SDU allocator
- * @param sdu_emit[in]    Callback of SDU emitter
+ * @param hdl[out]         Handle to new sink
+ * @param handle[in]       Connection handle
+ * @param pdus_per_sdu[in] Number of expected PDUs per SDU interval
+ * @param sdu_alloc[in]    Callback of SDU allocator
+ * @param sdu_emit[in]     Callback of SDU emitter
  * @return ISOAL_STATUS_OK if we could create a new sink; otherwise ISOAL_STATUS_ERR_SINK_ALLOC
  */
 isoal_status_t isoal_sink_create(
 	isoal_sink_handle_t      *hdl,
-	struct ll_conn_iso_stream *cis,
+	uint16_t                 handle,
+	uint8_t                  pdus_per_sdu,
```

Comment:
Can you confirm that `pdus_per_sdu` also covers the framed case? I was under the impression that the framed case would need some more nuance.

Reply:
I do believe you are right, I am working on implementing framed case next, might have to refactor some of the interfaces accordingly

Reply:
"A framed PDU with non-zero length shall contain one or more segments. A
framed PDU with zero length can be used as a padded PDU. Padding is
required when the data does not add up to the configured number of PDUs that
are specified by the BN parameter per BIS or CIS event."

Reply:
This seems to refer to padding at the event level, not the number of PDUs that constitute and SDU.

Reply:
I refactored a wider interface to isoal_sink_create() to carry over BN, sdu_interval and iso_interval. The number of expected PDUs per interval (for unframed case ) is now calculated inside isoal_sink_create(). Note, when framed case is implemented the parameters should now be available for similar calculations.